data/blog/intro-to-entity-framework-core-in-net-6.md

```diff
@@ -0,0 +1,586 @@
+    }
+
+    public Product? Get(int id)
+    {
+        return _dbContext.Products.Find(id);
+    }
+
+    public IReadOnlyList<Product> GetAll()
+    {
+        return _dbContext.Products.ToArray();
+    }
+
+    private Product? Update(Product product)
+    {
+        _dbContext.Products.Update(product);
+        _dbContext.SaveChanges();
+        return product;
+    }
+
+    public Product? Update(int id, decimal newPrice)
+    {
+        Product? product = Get(id);
+        if (product == null)
+        {
+            return null;
+        }
+        product.Price = newPrice;
+        return Update(product);
+    }
+}
+```
+
+Last but not least, we add some Utility classes in the root `Utilities` folder
+
+### Database.cs
+
+```cs
+using Ef_Intro.Data;
+namespace Ef_Intro.Utilities;
+
+public static class Database
+{
+    public static void InitDatabase(AppDbContext dbContext)
+    {
+        bool isEmpty = dbContext.Products.Any();
+        if (!isEmpty)
+        {
+            dbContext.Products.AddRange(
+                new() { Name = "Laptop", Price = 1000, Quantity = 50 },
+                new() { Name = "Mouse", Price = 50, Quantity = 25 },
+                new() { Name = "Keyboard", Price = 100, Quantity = 10 }
+            );
+
+            dbContext.SaveChanges();
+        }
+    }
+}
+```
+
+### Display.cs
+
+```cs
+using Ef_Intro.Data;
+using Ef_Intro.Entities;
+
+namespace Ef_Intro.Utilities;
+
+public static class Display
+{
+    public static void Print(string message)
+    {
+        Console.WriteLine(message);
+    }
+
+    public static void Products(IEnumerable<Product> products)
+    {
+        foreach (var product in products)
+        {
+            Console.WriteLine("Id: " + product.Id);
+            Console.WriteLine("Name: " + product.Name);
+            Console.WriteLine("Price: " + product.Price);
+            Console.WriteLine("Quantity: " + product.Quantity);
+            Console.WriteLine("===============================");
+        }
+    }
+
+    public static int Menu()
+    {
+        Console.WriteLine("1. Print products");
+        Console.WriteLine("2. Add product");
+        Console.WriteLine("3. Update product price");
+        Console.WriteLine("4. Delete product");
+        Console.WriteLine("5. Exit");
+        Console.Write("Enter your choice: ");
+        if (int.TryParse(Console.ReadLine(), out int choice))
+        {
+            return choice;
+        }
+
+        return -1;
+    }
+
+    public static void ResetConsole()
+    {
+        Console.WriteLine("\nPress any key to continue . . .");
+        Console.ReadKey();
+        Console.Clear();
+    }
+
+    internal static void InvalidInput()
+    {
+        Console.WriteLine("The input you submitted is not in a valid format");
+        Console.WriteLine("\nPress any key to continue . . .");
+        Console.ReadKey();
+        Console.Clear();
+    }
+}
+```
+
+The entry point finally looks like.
+
+### `Program.cs`
```

Comment:
There's a formatting inconsistency here, why the backticks?

Reply:
You are right there is inconsistency